drivers/npmx/npmx_shell.c

```diff
@@ -3417,164 +3425,157 @@ static npmx_timer_mode_t timer_mode_int_convert_to_enum(uint32_t timer_mode)
-
-	if (check_error_code(shell, err_code)) {
-		if (config_type == TIMER_CONFIG_TYPE_MODE) {
-			shell_print(shell, "Value: %d.", timer_config.mode);
-		} else if (config_type == TIMER_CONFIG_TYPE_PRESCALER) {
-			shell_print(shell, "Value: %d.", timer_config.prescaler);
-		} else if (config_type == TIMER_CONFIG_TYPE_COMPARE) {
-			shell_print(shell, "Value: %d.", timer_config.compare_value);
-		}
-	} else {
-		shell_error(shell, "Error: unable to read timer configuration.");
-	}
-
-	return 0;
-}
-
 static int timer_config_set(const struct shell *shell, size_t argc, char **argv,
-			    timer_config_type_t config_type)
+			    timer_config_param_t config_type)
 {
-	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
-
-	if (npmx_instance == NULL) {
-		shell_error(shell, "Error: shell is not initialized.");
-		return 0;
-	}
-
-	char config_name[10];
-	if (config_type == TIMER_CONFIG_TYPE_MODE) {
-		strcpy(config_name, "mode");
-	} else if (config_type == TIMER_CONFIG_TYPE_PRESCALER) {
-		strcpy(config_name, "prescaler");
-	} else if (config_type == TIMER_CONFIG_TYPE_COMPARE) {
-		strcpy(config_name, "period");
-	} else {
-		shell_error(shell, "Error: invalid config type value.");
-		return 0;
+	char *config_name;
+	switch (config_type) {
+	case TIMER_CONFIG_PARAM_MODE:
+		config_name = "mode";
+		break;
+	case TIMER_CONFIG_PARAM_PRESCALER:
+		config_name = "prescaler";
+		break;
+	case TIMER_CONFIG_PARAM_COMPARE:
+		config_name = "compare";
+		break;
 	}
 
-	if (argc < 2) {
-		shell_error(shell, "Error: missing timer %s value.", config_name);
+	args_info_t args_info = { .expected_args = 1,
+				  .arg = {
+					  [0] = { SHELL_ARG_TYPE_UINT32_VALUE, config_name },
+				  } };
+	if (!arguments_check(shell, argc, argv, &args_info)) {
 		return 0;
 	}
 
-	int err = 0;
-	uint32_t input_value = (uint32_t)shell_strtoul(argv[1], 0, &err);
-	if (err != 0) {
-		shell_error(shell, "Error: timer %s has to be an integer.", config_name);
+	npmx_timer_t *timer_instance = timer_instance_get(shell);
+	if (timer_instance == NULL) {
 		return 0;
 	}
 
-	npmx_timer_t *timer_instance = npmx_timer_get(npmx_instance, 0);
-
 	npmx_timer_config_t timer_config;
 	npmx_error_t err_code = npmx_timer_config_get(timer_instance, &timer_config);
-
 	if (!check_error_code(shell, err_code)) {
-		shell_error(shell, "Error: unable to read timer configuration.");
+		print_get_error(shell, "timer config");
 		return 0;
 	}
 
-	if (config_type == TIMER_CONFIG_TYPE_MODE) {
-		timer_config.mode = timer_mode_int_convert_to_enum(input_value);
-
+	uint32_t config_value = args_info.arg[0].result.uvalue;
+	switch (config_type) {
+	case TIMER_CONFIG_PARAM_MODE:
+		timer_config.mode = timer_mode_int_convert_to_enum(shell, config_value);
 		if (timer_config.mode == NPMX_TIMER_MODE_INVALID) {
-			shell_error(shell,
-				    "Error: timer mode can be: 0-boot monitor, 1-watchdog warning, "
-				    "2-watchdog reset, 3-general purpose or 4-wakeup.");
 			return 0;
```

Comment:
Why there is no error message printed here?

Reply:
It is moved to `timer_mode_int_convert_to_enum`.